src/librustc_mir/transform/mod.rs

```diff
@@ -265,6 +266,8 @@ fn mir_validated(
         &[&[
             // What we need to run borrowck etc.
             &promote_pass,
+            // FIXME(richkadel): is this the best place for the InstrumentCoverage pass?
```

Comment:
I think we probably want this to run in the `post_borrowck_cleanup` group.

Reply:
Done.

Note, when I moved the pass, I noticed that all of the `post_borrowck_cleanup` passes get called at least 3 times, when a macro like println!() is in the function body. 

It seems inefficient compared with where I had the pass before (which always only executed it once).

It still seems to work correctly (only codegen'ing the counter intrinsic once per function).

Is there a reason why these passes need to run multiple times? (And just making sure that reasoning also applies to the InstrumentCoverage pass.)

Reply:
See comments below regarding moving this to `mir_validated()`.